File: infrastructure/spring-boot-r2dbc-persistence-adapter/src/main/java/codes/fdk/blueprint/api/infrastructure/persistence/r2dbc/CategoryEntity.java
package codes.fdk.blueprint.api.infrastructure.persistence.r2dbc;

import codes.fdk.blueprint.api.domain.model.CategoryId;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;
import org.springframework.lang.Nullable;

@Table("categories")
record CategoryEntity(@Id @Nullable CategoryId id,
                      String name,
                      String slug,
                      @Nullable CategoryId parentId,
                      boolean visible) {}

File: infrastructure/vertx-postgres-persistence-adapter/src/main/kotlin/codes/fdk/blueprint/api/infrastructure/vertx/persistence/postgres/CategoryEntity.kt
package codes.fdk.blueprint.api.infrastructure.vertx.persistence.postgres

import codes.fdk.blueprint.api.domain.model.CategoryId

internal data class CategoryEntity(
    val id: CategoryId?,
    val name: String,
    val slug: String,
    val parentId: CategoryId?,
    val visible: Boolean
)
